src/config.c

```diff
@@ -274,3 +310,95 @@ int load_config(struct vpn_config *cfg, const char *filename)
 
 	return ret;
 }
+
+void destroy_vpn_config(struct vpn_config *cfg)
+{
+	free(cfg->pppd_log);
+	free(cfg->pppd_plugin);
+	free(cfg->pppd_ipparam);
+	free(cfg->pppd_ifname);
+	free(cfg->pppd_call);
+	free(cfg->ca_file);
+	free(cfg->user_cert);
+	free(cfg->user_key);
+	free(cfg->cipher_list);
+	while (cfg->cert_whitelist != NULL) {
+		struct x509_digest *tmp = cfg->cert_whitelist->next;
+		free(cfg->cert_whitelist);
+		cfg->cert_whitelist = tmp;
+	}
+}
+
+void merge_config(struct vpn_config *dst, struct vpn_config *src)
+{
+	if (src->gateway_host[0])
+		strcpy(dst->gateway_host, src->gateway_host);
+	if (src->gateway_port != invalid_cfg.gateway_port)
+		dst->gateway_port = src->gateway_port;
+	if (src->username[0])
+		strcpy(dst->username, src->username);
+	if (src->password[0])
+		strcpy(dst->password, src->password);
+	if (src->otp[0])
+		strcpy(dst->otp, src->otp);
+	if (src->realm[0])
+		strcpy(dst->realm, src->realm);
+	if (src->set_routes != invalid_cfg.set_routes)
+		dst->set_routes = src->set_routes;
+	if (src->set_dns != invalid_cfg.set_dns)
+		dst->set_dns = src->set_dns;
+	if (src->pppd_use_peerdns != invalid_cfg.pppd_use_peerdns)
+		dst->pppd_use_peerdns = src->pppd_use_peerdns;
+	if (src->use_syslog != invalid_cfg.use_syslog)
+		dst->use_syslog = src->use_syslog;
+	if (src->half_internet_routes != invalid_cfg.half_internet_routes)
+		dst->half_internet_routes = src->half_internet_routes;
```

Comment:
I noticed I'm missing `persistent` here for some reason (I probably started this on an old version), will fix

Reply:
(Note that persistent is set as an unsigned int in the config, read as a long int (boundaries only checked in cli option not in config file), but used as a bool?! Should this be changed to int like our other "bools" instead?)

Reply:
`persistent` is used as delay in seconds between connection attempts. persistent=300 would mean that _openfortivpn_ tries to re-establish a connection every 5 minutes after dropping out, and if I recall right it was on cli only at the beginning. Negative int values would overflow to very large uint numbers, which effectively would be a very long delay between connection attempts, in other words _openfortivpn_ would basically not reconnect within a reasonable time (which is the default anyway).

Reply:
Oh, I had missed it being used as `sleep` argument, right. I'll just add the overflow check to config.c